src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -211,23 +211,41 @@ private ObjectSyntax AddDecoratorsToBody(StatementSyntax statement, ObjectSyntax
         private void EmitParameter(JsonTextWriter jsonWriter, ParameterSymbol parameterSymbol, ExpressionEmitter emitter)
         {
             var declaringParameter = parameterSymbol.DeclaringParameter;
-            if (SyntaxHelper.TryGetPrimitiveType(declaringParameter) is not TypeSymbol primitiveType)
+
+            var properties = new List<ObjectPropertySyntax>();
+            if (parameterSymbol.Type is ResourceType resourceType)
+            {
+                // Encode a resource type as a string parameter with a metadata for the resource type.
+                properties.Add(SyntaxFactory.CreateObjectProperty("type", SyntaxFactory.CreateStringLiteral(LanguageConstants.String.Name)));
+                properties.Add(SyntaxFactory.CreateObjectProperty(
```

Comment:
The decision we made here is to move forward with this as an experimental feature to prevent bitrot. 

The big concern here is how we will specify more complex kinds of types. For example a union of a resource type and null. We will do more design work on the exact type representation in the interim.